testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/servlet/SAMLClockSkewAdapterTest.java

```diff
@@ -0,0 +1,145 @@
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.testsuite.adapter.servlet;
+
+import java.util.List;
+import javax.ws.rs.core.Response;
+import org.apache.http.util.EntityUtils;
+import org.hamcrest.Matcher;
+import org.jboss.arquillian.container.test.api.Deployer;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.keycloak.adapters.rotation.PublicKeyLocator;
+import org.keycloak.representations.idm.RealmRepresentation;
+import org.keycloak.testsuite.adapter.AbstractServletsAdapterTest;
+import org.keycloak.testsuite.adapter.filter.AdapterActionsFilter;
+import org.keycloak.testsuite.arquillian.AppServerTestEnricher;
+import org.keycloak.testsuite.arquillian.annotation.AppServerContainer;
+import org.keycloak.testsuite.util.DroneUtils;
+import org.keycloak.testsuite.util.SamlClientBuilder;
+import org.keycloak.testsuite.utils.arquillian.ContainerConstants;
+import org.keycloak.testsuite.utils.io.IOUtil;
+
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.not;
+import static org.keycloak.testsuite.adapter.AbstractServletsAdapterTest.samlServletDeployment;
+import static org.keycloak.testsuite.util.SamlClient.Binding.POST;
+
+
+@AppServerContainer(ContainerConstants.APP_SERVER_UNDERTOW)
+@AppServerContainer(ContainerConstants.APP_SERVER_WILDFLY)
+@AppServerContainer(ContainerConstants.APP_SERVER_WILDFLY_DEPRECATED)
+@AppServerContainer(ContainerConstants.APP_SERVER_EAP)
+@AppServerContainer(ContainerConstants.APP_SERVER_EAP6)
+@AppServerContainer(ContainerConstants.APP_SERVER_EAP71)
+//@AppServerContainer(ContainerConstants.APP_SERVER_TOMCAT7)
+//@AppServerContainer(ContainerConstants.APP_SERVER_TOMCAT8)
+//@AppServerContainer(ContainerConstants.APP_SERVER_TOMCAT9)
+//@AppServerContainer(ContainerConstants.APP_SERVER_JETTY92)
+//@AppServerContainer(ContainerConstants.APP_SERVER_JETTY93)
+//@AppServerContainer(ContainerConstants.APP_SERVER_JETTY94)
+public class SAMLClockSkewAdapterTest extends AbstractServletsAdapterTest {
+
+    private static final String CONTEXT_ROOT = "sales-post-clock-skew";
+    private static final String DEPLOYMENT_NAME_1 = CONTEXT_ROOT + "_1";
```

Comment:
Could the names (both the constants and method names) reflect the actual clock skew (e.g. `DEPLOYMENT_NAME_3_SEC`, `DEPLOYMENT_NAME_30_SEC`)?

Reply:
good idea, incorporated